src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms.TimeSeries;
+
+[assembly: LoadableClass(typeof(SrCnnEntireTransformer),
+        typeof(SrCnnEntireTransformer), null, typeof(SignatureLoadModel),
+        "SrCnnEntire Transformer",
+        SrCnnEntireTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(SrCnnEntireTransformer.SrCnnEntireModeler),
+        typeof(SrCnnEntireTransformer.SrCnnEntireModeler), null, typeof(SignatureLoadModel),
+        "SrCnnEntire Modeler",
+        SrCnnEntireTransformer.SrCnnEntireModeler.LoaderSignature)]
+namespace Microsoft.ML.Transforms.TimeSeries
+{
+    public sealed class SrCnnEntireTransformer : OneToOneTransformerBase
+    {
+        internal const string Summary = "This transformer detect timeseries anomalies for entire input using SRCNN";
+        internal const string LoaderSignature = "SrCnnEntireTransformer";
+        internal const string UserName = "SrCnn Entire Anomaly Detection";
+        internal const string ShortName = "srcnn entire";
+
+        internal sealed class Options : TransformInputBase
+        {
+            [Argument(ArgumentType.Required, HelpText = "The name of the source column.", ShortName = "src", SortOrder = 1, Purpose = SpecialPurpose.ColumnName)]
+            public string Source;
+
+            [Argument(ArgumentType.Required, HelpText = "The name of the target column.", ShortName = "tgt", SortOrder = 2)]
+            public string Target;
+
+            [Argument(ArgumentType.Required, HelpText = "The threshold to determine anomaly, score larger than the threshold is considered as anomaly. Must be in [0,1]. Default value is 0.3.",
+                ShortName = "thre", SortOrder = 101)]
+            public double Threshold = 0.3;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Divide the input data into batches to fit srcnn model. Must be -1 or a positive integer no less than 12. Default value is 1024.",
+                ShortName = "batch", SortOrder = 202)]
+            public int BatchSize = 1024;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The detection mode of SrCnnDetectMode enum type.",
+                ShortName = "mode", SortOrder = 303)]
+            public SrCnnDetectMode SrCnnDetectMode = SrCnnDetectMode.AnomalyOnly;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The sensitivity of boundaries.",
+                ShortName = "sens", SortOrder = 404)]
+            public double Sensitivity = 99;
+        }
+
+        internal readonly string InputColumnName;
+
+        internal readonly string OutputColumnName;
```

Comment:
`OneToOneTransformerBase` class has a field called `ColumnPairs` which contains this information.

Reply:
> `OneToOneTransformerBase` class has a field called `ColumnPairs` which contains this information.

Removed and refer to ColumnPairs for column name.